client/components/account-balances/hooks.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+type WPUser = {
+	id: number;
+	username: string;
+	name: string;
+	first_name: string;
+	nickname: string;
+	last_name: string;
+	email: string;
+	locale: string;
+};
+
+export const useCurrentWpUser = (): {
+	user?: WPUser;
+	displayName?: string;
+	isLoading: boolean;
+} => {
+	const { user, isLoading } = useSelect( ( select ) => {
+		const { getCurrentUser, isResolving } = select( 'core' );
+		return {
+			user: getCurrentUser() as WPUser | undefined,
+			isLoading: !! isResolving( 'getCurrentUser' ),
+		};
+	} );
+
+	const displayName = user?.first_name || user?.nickname || user?.name;
```

Comment:
Do you happen to know what the user.name property is here?

Theres a username, nick name, first name and name. 🤔

Reply:
`user.name` is the "Display name publicly as" value from `wp-admin/profile.php`.

Thinking more about the intention of this greeting, I think it is best to limit this to only return "First Name". The usage of `user.nickname` and `user.name` is likely to give undesirable results.

Reply:
> I think it is best to limit this to only return "First Name"
I've made this change in dd78aaa.

Reply:
> Thinking more about the intention of this greeting, I think it is best to limit this to only return "First Name". The usage of user.nickname and user.name is likely to give undesirable results.

Yeah good point. When I went and deleted my first name it came up with "Good morning, james 👋" - uncapitalised j. I can just imagine this will grind on peoples gears seeing their name incorrectly categorised.